patroni/__init__.py

```diff
@@ -138,18 +139,41 @@ def shutdown(self):
 
 
 def patroni_main():
+    logdir = os.environ.get('PATRONI_FILE_LOG_DIR', None)
+    logsize = os.environ.get('PATRONI_FILE_LOG_SIZE', 25000000)
+    lognum = os.environ.get('PATRONI_FILE_LOG_NUM', 4)
     logformat = os.environ.get('PATRONI_LOGFORMAT', '%(asctime)s %(levelname)s: %(message)s')
     loglevel = os.environ.get('PATRONI_LOGLEVEL', 'INFO')
     requests_loglevel = os.environ.get('PATRONI_REQUESTS_LOGLEVEL', 'WARNING')
-    logging.basicConfig(format=logformat, level=loglevel)
+
     logging.getLogger('requests').setLevel(requests_loglevel)
+    if not logdir:
+        logging.basicConfig(format=logformat, level=loglevel)
+    else:
+        root_logger = logging.getLogger()
+        root_logger.setLevel(loglevel)
+        handler = RotatingFileHandler(os.path.join(logdir, 'patroni.log'), mode='a', maxBytes=logsize,
+                                      backupCount=lognum)
+        handler.setFormatter(logging.Formatter(logformat))
+        root_logger.addHandler(handler)
+        logging.getLogger('requests').addHandler(handler)
 
     patroni = Patroni()
     try:
         patroni.run()
     except KeyboardInterrupt:
         pass
     finally:
+        root_handlers = logging.getLogger().handlers
```

Comment:
1. Flushing and closing of handlers should happen after `patroni.shutdown()`
2. There is the `logging.shutdown()` function, which seems already does what you want.